Model/Serverside/Data/Cart.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright © PureClarity. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+namespace Pureclarity\Core\Model\Serverside\Data;
+
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Customer\Model\Session as CustomerSession;
+use Magento\Quote\Model\Quote\Item;
+
+/**
+ * Serverside Cart handler, gets cart contents and determines if an event needs to be fired
+ */
+class Cart
+{
+    /** @var CheckoutSession */
+    private $checkoutSession;
+
+    /** @var CustomerSession */
+    private $customerSession;
+
+    /**
+     * @param CheckoutSession $checkoutSession
+     * @param CustomerSession $customerSession
+     */
+    public function __construct(
+        CheckoutSession $checkoutSession,
+        CustomerSession $customerSession
+    ) {
+        $this->checkoutSession = $checkoutSession;
+        $this->customerSession = $customerSession;
+    }
+
+    /**
+     * Checks to see if a basket event needs to be sent
+     *
+     * @return array
+     */
+    public function checkCart()
+    {
+        $lastCart = $this->customerSession->getPureclarityLastCartHash();
+        $cart = $this->getCartContents();
+        $send = false;
+
+        if (!$cart || $cart['hash'] !== $lastCart) {
+            $this->customerSession->setPureclarityLastCartHash($cart['hash']);
+            $send = true;
```

Comment:
will this always send if the cart is empty?

Reply:
hmm, actually, I think !$cart is redundant, as "getCartContents" always returns an array with 2 keys. I'll remove it as it's not needed.

Reply:
ah hang on, i think it should be there, but $lastCart instead of $cart (i.e. if we've not sent a basket in this session yet)